optaplanner-core/src/main/java/org/optaplanner/core/api/domain/solution/drools/ProblemFactCollectionProperty.java

```diff
@@ -20,24 +20,25 @@
 import java.lang.annotation.Target;
 import java.util.Collection;
 
-import org.kie.api.runtime.KieSession;
 import org.optaplanner.core.api.domain.entity.PlanningEntity;
 import org.optaplanner.core.api.domain.solution.PlanningSolution;
-import org.optaplanner.core.impl.score.director.drools.DroolsScoreDirector;
+import org.optaplanner.core.api.score.stream.ConstraintFactory;
+import org.optaplanner.core.api.score.stream.ConstraintProvider;
 import org.optaplanner.core.impl.solver.ProblemFactChange;
 
-import static java.lang.annotation.ElementType.*;
-import static java.lang.annotation.RetentionPolicy.*;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
 
 /**
  * Specifies that a property (or a field) on a {@link PlanningSolution} class is a {@link Collection} of problem facts.
- * A problem fact must not change during solving (except through {@link ProblemFactChange} event).
+ * A problem fact must not change during solving (except through a {@link ProblemFactChange} event).
  * <p>
- * The problem facts will be added as facts in the {@link KieSession} of the {@link DroolsScoreDirector},
- * so the score rules can use them.
+ * A {@link ConstraintProvider} builds constraints from such problem facts
+ * with {@link ConstraintFactory#from(Class)}.
  * <p>
  * Do not annotate {@link PlanningEntity planning entities} as problem facts:
- * they are automatically inserted into the {@link KieSession}.
+ * they are automatically available as facts with {@link ConstraintFactory#from(Class)}.
```

Comment:
This is not ideal either way. The original Javadoc assumed that we always use DRL, this one assumes we always use Constraint Streams. Maybe it should be more generic?

Reply:
Given the philosophy "ConstraintStreams First, Score DRL Second", we can only skip Score DRL if it's not crucial. This is indeed a crucial concept - you're right. Let me see what I can make of it.

Reply:
I just hate importing KieSession here. I want to avoid that.